Application/client/src/index.html

```diff
@@ -78,43 +78,103 @@ <h1 class="mb-4">Tune Tracer</h1>
                         };
                         responseDiv.appendChild(downloadMidiBtn);
 
-                        const pdfButton = document.createElement("button");
-                        pdfButton.innerText = "Get PDF";
-                        pdfButton.onclick = function() {
-                            pdfDiv.innerHTML = "";
-
-                            //create form data to post
-                            let postBody = new FormData();
-                            postBody.append('midiId', data.midiId);
-                            postBody.append('instrument', pred);
-
-                            fetch(`/getPdf`, {
-                                method: 'POST',
-                                body: postBody,
-                            })
-                                .then(response => response.blob())
-                                .then(blob => {
-                                    const blobUrl = URL.createObjectURL(blob);
-                                    const pdfObject = document.createElement("object");
-                                    pdfObject.data = blobUrl;
-                                    pdfObject.type = "application/pdf";
-                                    pdfObject.width = "100%";
-                                    pdfObject.height = "500px"; // Adjust size as needed
-                                    pdfDiv.appendChild(pdfObject);
-                                })
-                                .catch(error => console.error("error: ",error));
-                        };
-                        responseDiv.appendChild(pdfButton);
+                        // Automatically start PDF generation
+                        generatePdf(data.midiId, data.instrumentPrediction, pdfDiv);
                     }
                 })
                 .catch(error => {
-                  console.error('Error:', error);
-                  spinner.style.display = 'none';
+                console.error('Error:', error);
+                spinner.style.display = 'none';
                 });
             } else {
                 alert("Please select a file first.");
             }
+        }
+
+function uploadFile() {
```

Comment:
Is there a reason for having 2 uploadFile() functions (see line 40)

Reply:
Whoops! Fixed.